### ISSUE TITLE
New DSC-resource: AADAdministrativeUnit

### ISSUE THREAD

**Issue by salbeck-sit** [microsoft/Microsoft365DSC#2469]
### Description
New DSC resource for managing AAD Administrative Units

### Proposed properties
Id
Description
DisplayName
Visibility
MembershipType
MembershipRuleProcessingState
Members
ScopedRoleMembers
(+ standard properties)

### Special considerations or limitations
I've decided to *omit* support for Extensions - at least for now. One thing is that the format is tricky (arbitrary property-names with arbitrary values - perhaps feasible using MSFT_KeyValuePair?), another is that required permissions would include users, groups and devices since an update to Extensions of an AU would require properties to propagate to every user, group and/or device in scope for that AU.
Then there is MembershipType and MembershipRuleProcessingState. These are features of a public preview but I've chosen to include them as we use that preview in several places and it's been in preview for so long that I believe it's a matter of weeks before it's GA.

**Comment by salbeck-sit** [comment]
I have built the resource and am in the process of ensuring that unit-tests are comprehensive enough to validate functionality.

**Comment by NikCharlebois** [comment]
@salbeck-sit I actually had cycles yesterday morning and did a first stab at it. It currently only supports configuringdynamic memberships, but reports on assigned permissions. This resource had proven to be more complex than anticipated to build. Let's compare and merge in whatever additional properties you are handling with yours.

**Comment by salbeck-sit** [comment]
@NikCharlebois I also included code for ScopedRoleMember and some preliminary thoughts on Extensions that led me to exclude the attribute. I can't do a PR on top of your PR but have a look at https://github.com/salbeck-sit/Microsoft365DSC/tree/AADAdministrativeUNit/Modules/Microsoft365DSC/DSCResources/MSFT_AADAdministrativeUnit
PS When I built it, I think I found a flaw in DRG when it builds code for Test-TargetResource for handling comparisons of attributes where one is empty. I'll see if I can find it again.
